src/odemis/driver/zeiss.py

```diff
@@ -513,66 +560,101 @@ def _refreshPosition(self):
+        :param future: a future that can be used to manage a move
+        :param shift (dict): contains float position values in m
         """
-        x, y, z, _ = self.parent.GetStagePosition()
+        # set the values of these axes to 0.0
+        # if none of these are used in the move the command for 3 axes will be used
+        t = r = m = 0.0
+
+        # get the stage position for all axes
+        x, y, z, t, r, m, _ = self.parent.GetStagePosition6()
+
+        target_pos = {}
+
         if "x" in shift:
             x += shift["x"] * 1e3
+            target_pos["x"] = x * 1e-3
         if "y" in shift:
             y += shift["y"] * 1e3
+            target_pos["y"] = y * 1e-3
         if "z" in shift:
             z += shift["z"] * 1e3
-
-        target_pos = self._applyInversion({"x": x * 1e-3, "y": y * 1e-3, "z": z * 1e-3})
-        # Check range (for the axes we are moving)
+            target_pos["z"] = z * 1e-3
+        if "rx" in shift:
+            t += shift["rx"]
+            target_pos["rx"] = t
+        if "rz" in shift:
+            r += shift["rz"]
+            target_pos["rz"] = r
+        if "m" in shift:
+            m += shift["m"] * 1e3
+            target_pos["m"] = m * 1e-3
+
+        target_pos = self._applyInversion(target_pos)
+        # Check range for axes which have finite ranges (for the axes we are moving)
         for an in shift.keys():
             rng = self.axes[an].range
             p = target_pos[an]
-            if not rng[0] <= p <= rng[1]:
-                raise ValueError("Relative move would cause axis %s out of bound (%g m)" % (an, p))
+            if not util.almost_equal(rng[1] - rng[0], 2 * math.pi):
+                if not rng[0] <= p <= rng[1]:
+                    raise ValueError("Relative move would cause axis %s out of bound (%g m)" % (an, p))
 
-        self._moveTo(future, x, y, z)
+        self._moveTo(future, x, y, z, t, r, m)
 
     def _doMoveAbs(self, future, pos):
         """
-        move to position pos
-        pos (float): unit m
+        prepare the stage for an absolute move using the values in pos
+        :param future: a future that can be used to manage a move
+        :param pos (dict): contains float position values in m
         """
+        # set the values of these axes to 0.0
+        # if none of these are used in the move the command for 3 axes will be used
+        t = r = m = 0.0
 
-        # Don't change position for unspecified coordinates
-        x, y, z, _ = self.parent.GetStagePosition()
-        # Convert to mm
+        # get the stage position for all axes
+        x, y, z, t, r, m, _ = self.parent.GetStagePosition6()
+
+        # Convert the linear axes to mm
         if "x" in pos:
             x = pos["x"] * 1e3
         if "y" in pos:
             y = pos["y"] * 1e3
         if "z" in pos:
             z = pos["z"] * 1e3
+        if "rx" in pos:
+            t = pos["rx"]
+        if "rz" in pos:
+            r = pos["rz"]
```

Comment:
I've changed the input in the testcases from radians to degree already. So this can stay this way, like it should be for the real case command which expects degree positions.